Adds missing unit tests

p-galligan opened the pull request:
## Changes
Fixes #70 . Gets us to 100% coverage for the data_helpers and serializers. 
also fixes #81

Currently no coverage of the decorator function, but that maybe should be its own issue.

Commit:
expand coverage

expand coverage to fully cover get_expression function

Commit:
add rights_statement fixture

adding a standalone rights statement fixture

Commit:
add coverage to tox

adding coverage to tox file

Commit:
update coverage

Updates coverage to only run against current directory

Commit:
update tox

Updating tox to only run coverage against rac_aspace directory. Also adds a fixture

Commit:
improve format_resource_id coverage

Updating format_resource_id helper and test_format_resource_id for better test coverage.

Commit:
update is_restricted test

update is_restricted test to catch rights statements when no note is present

Commit:
updates get_expression and test_format_from_ob

Updating test_format_form_obg and get_expression and tests

Commit:
add fixtures and strings for testing

Adds fixtures and strings for testing. These tests are currently failing.

Commit:
update tests

updates test coverage for is_restricted and indicates_restriction

Commit:
improve coverage format_from_object

Improves coverage to catch exception

helrond committed:
rename fixtures

helrond committed:
debugs notes

helrond committed:
fixing merge conflict

helrond committed:
Merge pull request #82 from RockefellerArchiveCenter/issue-70-bugfix

Fixes failing tests on notes

helrond commented on tests/test_data_helpers.py:
```diff
@@ -122,14 +147,24 @@ def test_strip_html_tags(self):
+        ]:
+            date = self.obj_from_fixture(fixture)
+            if format_string is None:
+                self.assertRaises(
+                    Exception, data_helpers.format_from_obj, date, None)
```
```suggestion
                self.assertRaises(
                    Exception, data_helpers.format_from_obj, date, None)
```
I'd be interested in thoughts on this, but I've always seen this done like:

```
with self.assertRaises(Exception):
    data_helpers.format_from_obj(date, format_string)
```
Not sure what best practice is though...

p-galligan replied:
So, I don't really know best practice either.

From reading [the documentation](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises), it feels like you can do either, but I'm not sure what the advantage of writing the code inline rather than a function is. Given that we were passing args, I thought the function method was the preferred way due to documentation.

helrond replied:
From my understanding it seems like you'd want to use the `with` block if you wanted to assign that exception to a variable so you can inspect it (see line 162). Given that this is raising a very generic `Exception` I think it might be wise to use a similar pattern to the below to check the exception message? Thoughts?

p-galligan replied:
Sure, that makes sense to me. I'll make the changes.

helrond commented on tests/test_data_helpers.py:
```diff
@@ -122,14 +147,24 @@ def test_strip_html_tags(self):
+            ("date_expression.json", None),
+        ]:
+            date = self.obj_from_fixture(fixture)
+            if format_string is None:
```
```suggestion
            if not format_string:
```
Can we just check like that?

helrond commented on tests/test_data_helpers.py:
```diff
@@ -31,10 +31,25 @@ def obj_from_fixture(self, filename, client=None):
+                ("note_single.json", ["New York Mets"])]:
+            note = self.obj_from_fixture(fixture)
+            result = data_helpers.get_note_text(note)
+            print(result)
```
```suggestion
            print(result)
```
Let's remove this line, eh?

helrond reviewed (changes requested):
This looks really nice. Just requested some minor changes.

Commit:
update tests

Removing a print statement and changing the diction for checking if something is None.

Commit:
update format_from_object test

Updates assertRaises syntax

helrond commented on tests/test_data_helpers.py:
```diff
@@ -122,14 +146,24 @@ def test_strip_html_tags(self):
+            date = self.obj_from_fixture(fixture)
+            if not format_string:
+                with self.assertRaises(Exception):
+                    data_helpers.format_from_obj(date, format_string)
```
```suggestion
                with self.assertRaises(Exception) as excpt:
                    data_helpers.format_from_obj(date, format_string)
                self.assertEqual("No format string provided.", str(excpt.exception)) 
```
Let's target the exception message since we know what it should be.

Commit:
target exception

Targeting exception and formatting for autopep8

HaSistrunk reviewed (approved):
Nice!
